src/index.js

```diff
@@ -47,11 +48,13 @@ export default function babel ( options ) {
 			if ( id === HELPERS ) {
 				const pattern = new RegExp( `babelHelpers\\.(${keywordHelpers.join('|')})`, 'g' );
 
-				const helpers = buildExternalHelpers( externalHelpersWhitelist, 'var' )
+				let helpers = buildExternalHelpers( externalHelpersWhitelist, 'var' )
 					.replace( pattern, 'var _$1' )
 					.replace( /^babelHelpers\./gm, 'export var ' ) +
 					`\n\nexport { ${keywordHelpers.map( word => `_${word} as ${word}`).join( ', ')} }`;
-
+				// Apply babel transforming on helpers, in case ES5/ES3 transform plugins are set in options.
+				helpers = transform( helpers, originalOptions ).code;
```

Comment:
when you ditch copying `options` on the top of the plugin you will have to just pass `options` here instead of `originalOptions`